### PR TITLE
added: cli arguments for both the port and host

### PR DESCRIPTION
**Pull request by zt4ff**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -138,6 +138,10 @@ def log_startup_message(port_num, is_https_enabled):
                 SSL_CERT = sys.argv[i+1]
             elif (arg == "--ssl-key" and (i+1) < (len(sys.argv))):
                 SSL_KEY = sys.argv[i+1]
+            elif (arg == "--port" and (i+1) < (len(sys.argv))):
+                port = int(sys.argv[i+1])
+            elif (arg == "--host" and (i+1) < (len(sys.argv))):
+                host = int(sys.argv[i+1])
```
Comment (shubham1172) [inline main.py:144]: `host` is not an `int`. See line 129.

File: main.py [hunk 2]
```diff
@@ -138,6 +138,10 @@ def log_startup_message(port_num, is_https_enabled):
                 SSL_CERT = sys.argv[i+1]
             elif (arg == "--ssl-key" and (i+1) < (len(sys.argv))):
                 SSL_KEY = sys.argv[i+1]
+            elif (arg == "--port" and (i+1) < (len(sys.argv))):
+                port = int(sys.argv[i+1])
```
Comment (shubham1172) [inline main.py:142]: Please handle converting string to int and avoid runtime issues. You can catch it gracefully, log, and exit.

**Review by shubham1172 — CHANGES_REQUESTED** [review]
Hello @zt4ff, thank you for your PR, requested some changes!

**Comment by zt4ff** [comment]
@shubham1172, I resolved the comments. Thanks.

File: main.py [hunk 3]
```diff
@@ -138,6 +138,14 @@ def log_startup_message(port_num, is_https_enabled):
                 SSL_CERT = sys.argv[i+1]
             elif (arg == "--ssl-key" and (i+1) < (len(sys.argv))):
                 SSL_KEY = sys.argv[i+1]
+            elif (arg == "--port" and (i+1) < (len(sys.argv))):
+                try:
+                    port = int(sys.argv[i+1])
+                except:
```
Comment (shubham1172) [inline main.py:144]:
```suggestion
                except ValueError:
```

File: main.py [hunk 4]
```diff
@@ -138,6 +138,14 @@ def log_startup_message(port_num, is_https_enabled):
...
+                    exit("The --port argument should be a number")
```
Comment (shubham1172) [inline main.py:145]:
```suggestion
                    console.log(f"Error: invalid --port, {verr}, exiting", color="r")
                    sys.exit(1)
```

**Comment by zt4ff** [comment]
updated

File: main.py [hunk 5]
```diff
@@ -138,6 +138,15 @@ def log_startup_message(port_num, is_https_enabled):
                 SSL_CERT = sys.argv[i+1]
             elif (arg == "--ssl-key" and (i+1) < (len(sys.argv))):
                 SSL_KEY = sys.argv[i+1]
+            elif (arg == "--port" and (i+1) < (len(sys.argv))):
+                try:
+                    port = int(sys.argv[i+1])
+                except ValueError as verr:
+                    console.log(f"Error: invalid --port, {verr}, exiting", color="r")
+                    sys.exit(1)
+
```
Comment (shubham1172) [inline main.py:147]:
```suggestion
```

File: main.py [hunk 6]
```diff
@@ -138,6 +138,14 @@ def log_startup_message(port_num, is_https_enabled):
                 SSL_CERT = sys.argv[i+1]
             elif (arg == "--ssl-key" and (i+1) < (len(sys.argv))):
                 SSL_KEY = sys.argv[i+1]
+            elif (arg == "--port" and (i+1) < (len(sys.argv))):
+                try:
+                    port = int(sys.argv[i+1])
+                except ValueError as verr:
+                    console.log(f"Error: invalid --port, {verr}, exiting", color="r")
```
Comment (shubham1172) [inline main.py:145]:
```suggestion
                    console.log(
                        f"Error: invalid --port, {verr}, exiting", color="r")
```

**Review by shubham1172 — APPROVED** [review]
Thank you @zt4ff for your contributions.

To many more contributions! 🎉

**Comment by zt4ff** [comment]
closes #40

**Pull request merged**